.ci/pitest.sh

```diff
@@ -38,13 +38,21 @@ pitest-annotation|pitest-design \
 |pitest-packagenamesloader \
 |pitest-common-2|pitest-misc|pitest-xpath \
 |pitest-filters \
-|pitest-main \
-|pitest-coding)
+|pitest-main)
   mvn -e -P$1 clean test org.pitest:pitest-maven:mutationCoverage;
   declare -a ignoredItems=();
   checkPitestReport "${ignoredItems[@]}"
   ;;
 
+pitest-coding)
+  mvn -e -P$1 clean test org.pitest:pitest-maven:mutationCoverage;
+  declare -a ignoredItems=(
+  "RedundantThisCheck.java.html:<td class='covered'><pre><span  class='survived'>                    register(ast, frameStack.poll());</span></pre></td></tr>"
+  "RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>                    register(ast, frameStack.poll());</span></pre></td></tr>"
```

Comment:
I think it should be fixed, not just added to exclusions.
I guess it was not here before, so there is some issue with tests

Reply:
I actually tried to fix it. I've done regression on the mutation and I did find cases where exceptions where thrown which I've added to UTs.
But somehow pitest won't pass. I was looking to discuss with you on this matter.
Because if you remove that code portion and run `mvn verify` will fail. 
So I can't understand why pitest still report it as not necessary.

Besides, what's even more strange is that, before changes, no pitest failure was reported inside `RequireThisCheck`, all I did for that line is replacing
```java
frames.put(ast, frameStack.poll()); // (1)
```
by 
```
register(ast, frameStack.poll()); // (2)
```
You can check by yourself that the `register` method does nothing but `(1)`.

So how is it possible that `(1)` passes pitest but not `(2)` ?

Reply:
I think I faced this problem when pitest was not okay with replacing put() with method call. 
The only reason I can imagine is that `put` is not a void method while replaced method is void. And pitest has some mutation that removes void methods.

Reply:
See my comment for `InputRedundantThisAnonymousObjects` to solve this issue.

Reply:
@wilcoln , please try to refactor a bit this code and move nested execution of `poll` to separate statement.